Source/WebCore/editing/DeleteSelectionCommand.cpp

```diff
@@ -609,8 +609,7 @@ void DeleteSelectionCommand::makeStylingElementsDirectChildrenOfEditableRootToPr
     auto nodes = intersectingNodes(*range).begin();
     while (nodes) {
         Ref node = *nodes;
-        auto shouldMove = is<HTMLLinkElement>(node)
-            || (is<HTMLStyleElement>(node) && !downcast<HTMLStyleElement>(node.get()).hasAttributeWithoutSynchronization(scopedAttr));
```

Comment:
I'm not great at boolean logic, but thankfully there's tests. Will fix this now.